internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -162,7 +177,35 @@ func convertDestinationRule(allPorts []string, dr *pbauth.DestinationRule) ([]st
 		out = append(out, p)
 	}
 
-	return out, DestinationRule{}
+	if len(dr.String()) == 0 {
+		return out, nil
+	}
+
+	psdr := &pbproxystate.DestinationRule{
+		PathExact:  dr.PathExact,
+		PathPrefix: dr.PathPrefix,
+		PathRegex:  dr.PathRegex,
+		Methods:    dr.Methods,
+	}
+	hrs := make([]*pbproxystate.DestinationRuleHeader, len(dr.Headers))
+	for i, hr := range dr.Headers {
+		hrs[i] = &pbproxystate.DestinationRuleHeader{
+			Name:    hr.Name,
+			Present: hr.Present,
+			Exact:   hr.Exact,
+			Prefix:  hr.Prefix,
+			Suffix:  hr.Suffix,
+			Regex:   hr.Regex,
+			Invert:  hr.Invert,
+		}
+	}
+	psdr.DestinationRuleHeader = hrs
+
+	if len(psdr.String()) > 0 {
+		return out, psdr
+	}
```

Comment:
I think it's fine in this case, but it's not typically safe to rely on `String` returning `""` meaning the struct is empty.